core/src/main/java/org/apache/calcite/sql/validate/EmptyScope.java

```diff
@@ -133,6 +133,88 @@ class EmptyScope implements SqlValidatorScope {
     }
   }
 
+  @Override public void resolveSchema(List<String> names, SqlNameMatcher nameMatcher, Path path,
+      Resolved resolved) {
```

Comment:
For this section, I would look at resolveTable, since this function is very heavily base on that function.